File: nexobytes_edutech/lib/classes/student.dart
class Student {
  final String username;
  final String password;
  final String name;
  final String classId;
  final String className;
  final int createdAt;
  final String createdBy;

  Student({
     required this.username, required this.password, this.name="", required this.classId, this.className="", this.createdAt=0, this.createdBy=""
  });

  factory Student.fromJson(Map<String, dynamic> json) {
    return Student(
        username: json['username'],
        password: json['password'],
        name: json['name'],
        classId: json['classId'],
        className: json['className'],
        createdAt: json['createdAt'],
        createdBy: json['createdBy']
    );
  }

  Map<String, dynamic> toJson() => {
    'username': username,
    'password': password,
    'name': name,
    'classId': classId,
    'className': className,
    'createdAt': createdAt,
    'createdBy': createdBy
  };

}

File: server/models/student.js
const mongoose = require("mongoose");
const uniqueValidator = require('mongoose-unique-validator');

const Schema = mongoose.Schema;

const studentSchema = new Schema({
  username: { type: String, required: true, unique: true },
  classId: { type: String, required: true },
  className: { type: String, required: true },
  name: { type: String, required: true },
  password: { type: String, required: true },
  createdAt: { type: Number, required: true },
  createdBy: { type: String, required: true },
  isActive: { type: Boolean, required: true },
});

studentSchema.plugin(uniqueValidator);

module.exports = mongoose.model("Student", studentSchema);
